internal/provider/query_annotation_resource.go

```diff
@@ -0,0 +1,202 @@
+package provider
+
+import (
+	"context"
+	"errors"
+
+	"github.com/hashicorp/terraform-plugin-framework-validators/stringvalidator"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+
+	"github.com/honeycombio/terraform-provider-honeycombio/client"
+	"github.com/honeycombio/terraform-provider-honeycombio/internal/helper"
+	"github.com/honeycombio/terraform-provider-honeycombio/internal/helper/modifiers"
+	"github.com/honeycombio/terraform-provider-honeycombio/internal/models"
+)
+
+// Ensure the implementation satisfies the expected interfaces.
+var (
+	_ resource.Resource              = &queryAnnotationResource{}
+	_ resource.ResourceWithConfigure = &queryAnnotationResource{}
+)
+
+type queryAnnotationResource struct {
+	client *client.Client
+}
+
+func NewQueryAnnotationResource() resource.Resource {
+	return &queryAnnotationResource{}
+}
+
+func (*queryAnnotationResource) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_query_annotation"
+}
+
+func (r *queryAnnotationResource) Configure(_ context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	w := getClientFromResourceRequest(&req)
+	if w == nil {
+		return
+	}
+
+	c, err := w.V1Client()
+	if err != nil || c == nil {
+		resp.Diagnostics.AddError("Failed to configure client", err.Error())
+		return
+	}
+	r.client = c
+}
+
+func (*queryAnnotationResource) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Description: "Query annotations allow you to add notes and descriptions to specific queries for better documentation and context.",
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Description: "The unique identifier for this Query Annotation.",
+				Computed:    true,
```

Comment:
q - looks like the ApiKeyResource

https://github.com/honeycombio/terraform-provider-honeycombio/blob/448e063ba019b9f3cccb1f08f7b0387fa92f29ce/internal/provider/api_key_resource.go#L70-L71

specifies `Required: false` and `Optional: false` for `id`. I guess those are zero values so they default to false, but should we include them for clarity?

Reply:
Sure! clarity is nice